core/src/main/java/zipkin2/reporter/AsyncReporter.java

```diff
@@ -50,7 +33,8 @@
  *
  * @param <S> type of the span, usually {@link zipkin2.Span}
  */
-public abstract class AsyncReporter<S> extends Component implements Reporter<S>, Flushable {
+public final class AsyncReporter<S> extends Component implements Reporter<S>, Flushable {
```

Comment:
I think going from abstract to final here might be a breaking change (not sure if you're intending for those yet). Abstract to non-final probably isn't if it's well behaved in terms of random subclasses

Reply:
Because the constructor is package-private, I think it won't have any impact, right?

Reply:
I mean no one could have sub-classed it without sharing our package which wouldn't be supported anyway.

Reply:
I can take final off, but would want to make a comment. Would a reason be something like in case someone is doing bytecode instrumentation?

Reply:
made a comment anyway!